### PR TITLE
Feature/new color recognition system

### PR DESCRIPTION
**Pull request by zdzarsky**

### CODE REVIEW

File: mrc/localization/color/defaults.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from enum import Enum
+
+
+class DefaultColorValues(Enum):
+    """
+    Enum mapping color names into their hue (HSV color model) value from range [0, 360).
+    """
+    red = range(-30, 30)
+    yellow = range(30, 90)  # (100, 100, 0)
```
Comment (Lukasz1928) [inline mrc/localization/color/defaults.py]: How about adding information about encoding of commented colors for better readibility?

File: mrc/localization/color/quick_mean/quick_mean.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    def extract(self, image):
+        _image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+        return self._execute_analysis(_image)
+
+    def _execute_analysis(self, chunk):
+        analysis = self.analyser.analyse_chunk(chunk)
+        max_value = -1
+        final_color = None
+        for key, value in analysis.items():
```
Comment (Lukasz1928) [inline mrc/localization/color/quick_mean/quick_mean.py]: return max(tuple(analysis.items()), key=lambda x:x[1])[0] or sth similar, mr functional programming?

File: tests/localization/color/quick_mean/common.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import cv2
+
+from tests.test_utils.read_image import read_image
+
+image = cv2.imread('tests/localization/color/quick_mean/test_image.jpg')
+
+colors_analyzed = [
```
Comment (Lukasz1928) [inline tests/localization/color/quick_mean/common.py]: Why such values? Can we distinguish between e-163 and e-274 or similar?

File: mrc/localization/color/quick_mean/quick_mean.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+class Predicates:
+    allowed = {
+        'gaussian': GaussianHSVPredicate()
+    }
+
+
+class Analyser:
+    def __init__(self, color_values, predicate_strategy='gaussian'):
+        self._clusters = 1
```
Comment (Lukasz1928) [inline mrc/localization/color/quick_mean/quick_mean.py]: todelete

File: mrc/localization/color/quick_mean/quick_mean.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        mean_color = np.mean(pixel_list, axis=0)
+        return self._color_analysis(rgb2hsv(*mean_color))
+
+    def _color_analysis(self, hsv_pixel):
+        hue, _, _ = hsv_pixel
+        return {color_val.name: self._resolving_predicate(hue, color_val.value) for color_val in
+                self._color_values}
+
+
+class Extractor:
```
Comment (Lukasz1928) [inline mrc/localization/color/quick_mean/quick_mean.py]: Maybe rename to ColorCalculator or something similar? Extractor is not the most descriptive name I've seen.

File: mrc/localization/color/quick_mean/quick_mean.py (hunk 4 above)
Comment (Maheliusz) [inline mrc/localization/color/quick_mean/quick_mean.py]: delet dis

**Pull request merged**